[bit@0.2.51] Cannot find module '@yarnpkg/cli' when using yarn as package manager

<!--

Before submitting a new issue, please search past issues (open or closed).  

-->

### Describe the bug

Error occurs: Cannot find module '@yarnpkg/cli' when running `bit install` with package manager `yarn`

### Steps to Reproduce

1. run `bit install` 

### Screenshots, exceptions and logs

```bash
Cannot find module '@yarnpkg/cli'
Require stack:
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@yarnpkg/plugin-pack/lib/commands/pack.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@yarnpkg/plugin-pack/lib/index.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@teambit/yarn/node_modules/@yarnpkg/plugin-npm/lib/npmPublishUtils.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@teambit/yarn/node_modules/@yarnpkg/plugin-npm/lib/index.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@teambit/yarn/node_modules/@yarnpkg/cli/lib/tools/getPluginConfiguration.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@teambit/yarn/node_modules/@yarnpkg/cli/lib/index.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@teambit/yarn/dist/yarn.package-manager.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@teambit/yarn/dist/yarn.main.runtime.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@teambit/bit/dist/load-bit.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@teambit/bit/dist/app.js
- /home/fix777/.bvm/versions/0.2.51/bit-0.2.51/node_modules/@teambit/bit/bin/bit
```

### Specifications

- Bit version: 0.2.51
- Workspace type: harmony
- Node version: v18.17.1
- npm / yarn version: from bit
- Platform: elementary OS 7 Horus(Built on Ubuntu 22.04.3 LTS) Linux 6.2.6-76060206-generic

--- a/workspace.jsonc
+++ b/workspace.jsonc
@@ -844,6 +844,9 @@
             "@teambit/ui-foundation.ui.navigation.react-router-adapter": "6.1.1",
             "@apollo/client": "3.6.9",
             "@teambit/legacy": "1.0.559",
+            // These are the peer dependencies of Yarn
+            "@yarnpkg/cli": "3.6.1",
+            "@yarnpkg/core": "3.5.2",
             "@yarnpkg/plugin-pack": "3.2.0",
             "graphql": "15.8.0",
             "browserslist": "4.16.3",

